src/secrets-kubernetes/src/lib.rs

```diff
@@ -33,15 +56,64 @@ impl KubernetesSecretsController {
             },
         };
         let client = Client::try_from(kubeconfig)?;
+        let secret_api: Api<Secret> = Api::default_namespaced(client);
 
-        Ok(KubernetesSecretsController {
-            _kube_client: client,
-        })
+        let name: String = format!("{}", HARDCODED_NAME);
+
+        let secret = KubernetesSecretsController::make_secret_with_name(name.clone());
+        match secret_api.create(&PostParams::default(), &secret).await {
+            Ok(_) => Ok(()),
+            Err(e) => {
+                if e.to_string().contains("AlreadyExists") {
```

Comment:
I don't like this. Is there a better way to check this?

Reply:
You can match on the type of error https://docs.rs/kube/latest/kube/enum.Error.html . I think you want `Error::Api`, which has an error code.

```rust
match secret_api.create(&PostParams::default(), &secret).await {
    Ok(_) => Ok(()),
    Err(kube::Error::Api(e)) if e.code == already_exists_code => {
        // handle already exists case
    },
    Err(e) => {
        // handle real errors
    }
}
```

Reply:
thanks for the destructuring code :)